src/protocol/basics/reshare.rs

```diff
@@ -156,6 +181,45 @@ impl<'a, F: Field> Reshare<MaliciousContext<'a, F>, RecordId> for MaliciousRepli
     }
 }
 
+#[async_trait]
+/// Reshare algorithm for xor secret shares.
+impl<'a, B: Fp2Array> Reshare<SemiHonestContext<'a>, RecordId> for XorReplicated<B> {
+    async fn reshare<'fut>(
+        &self,
+        ctx: SemiHonestContext<'a>,
+        record_id: RecordId,
+        to_helper: Role,
+    ) -> Result<Self, Error>
+    where
+        SemiHonestContext<'a>: 'fut,
+    {
+        let r = ctx.prss().generate_bit_arrays(record_id);
+        let (s0, s1) = semi_honest_reshare(ctx, record_id, to_helper, self, r).await?;
+        Ok(XorReplicated::new(s0, s1))
+    }
+}
+
+#[async_trait]
+/// Malicious version of xor resharing is executed in semi-honest context.
+///
+/// As of March 2023, we aren't sure what is a proper way of doing reshare in malicious context.
+/// We believe that, even though an additive attack is possible, a malicious helper can ONLY
+/// corrupt the protocol output, but cannot learn any private info.
+impl<'a, F: Field, B: Fp2Array> Reshare<MaliciousContext<'a, F>, RecordId> for XorReplicated<B> {
+    async fn reshare<'fut>(
+        &self,
+        ctx: MaliciousContext<'a, F>,
+        record_id: RecordId,
+        to_helper: Role,
+    ) -> Result<Self, Error>
+    where
+        MaliciousContext<'a, F>: 'fut,
+    {
+        self.reshare(ctx.semi_honest_context(), record_id, to_helper)
+            .await
```

Comment:
```suggestion
        panic!("we don't know how to do this");
```

Reply:
I assume that our position might have changed regarding this due to a possible attack mentioned by Daniel and Ben Case, but the first intention was to run reshare using semi-honest context even in the malicious settings. If we insert panic here, we won't be able to run e2e malicious IPA, and we might just as well not land this PR for now?

Reply:
I only want to ensure that we track this issue properly.  Either we put the insecure code in and open an issue, or we ensure that we only deploy secure code.  I don't mind which.

Reply:
Opened #518.

Reply:
OK, I think I now know of a way how to do this. Let's land this PR with the `panic!()` call as @martinthomson has suggested, and then follow up with a diff to fix it.